src/mc_tasks/ImpedanceTask.cpp

```diff
@@ -0,0 +1,316 @@
+                         deltaCompVelLinLimit_);
+    deltaCompVelW_.linear().normalize();
+    deltaCompVelW_.linear() *= deltaCompVelLinLimit_;
+  }
+  if(deltaCompVelW_.angular().norm() > deltaCompVelAngLimit_)
+  {
+    mc_rtc::log::warning("angular deltaCompVel limited from {} to {}", deltaCompVelW_.angular().norm(),
+                         deltaCompVelLinLimit_);
+    deltaCompVelW_.angular().normalize();
+    deltaCompVelW_.angular() *= deltaCompVelAngLimit_;
+  }
+
+  if(deltaCompPoseW_.translation().norm() > deltaCompPoseLinLimit_)
+  {
+    mc_rtc::log::warning("linear deltaCompPose limited from {} to {}", deltaCompPoseW_.translation().norm(),
+                         deltaCompPoseLinLimit_);
+    deltaCompPoseW_.translation().normalize();
+    deltaCompPoseW_.translation() *= deltaCompPoseLinLimit_;
+  }
+  Eigen::AngleAxisd aaDeltaCompRot(deltaCompPoseW_.rotation());
+  if(aaDeltaCompRot.angle() > deltaCompPoseAngLimit_)
+  {
+    mc_rtc::log::warning("angular deltaCompPose limited from {} to {}", aaDeltaCompRot.angle(), deltaCompPoseAngLimit_);
+    aaDeltaCompRot.angle() = deltaCompPoseAngLimit_;
+    deltaCompPoseW_.rotation() = aaDeltaCompRot.toRotationMatrix();
+  }
+
+  // 4. Set compliance values to the targets of SurfaceTransformTask
+  refAccel(T_0_s * (desiredAccelW_ + deltaCompAccelW_)); // represented in the surface frame
+  refVelB(T_0_s * (desiredVelW_ + deltaCompVelW_)); // represented in the surface frame
+  target(compliancePose()); // represented in the world frame
+}
+
+void ImpedanceTask::reset()
+{
+  // Set the target pose of SurfaceTransformTask to the current pose
+  // Reset the target velocity and acceleration of SurfaceTransformTask to zero
+  SurfaceTransformTask::reset();
+
+  // Set the desired and compliance poses to the SurfaceTransformTask target (i.e., the current pose)
+  desiredPoseW_ = target();
+  deltaCompPoseW_ = sva::PTransformd::Identity();
+
+  // Reset the desired and compliance velocity and acceleration to zero
+  desiredVelW_ = sva::MotionVecd::Zero();
+  desiredAccelW_ = sva::MotionVecd::Zero();
+  deltaCompVelW_ = sva::MotionVecd::Zero();
+  deltaCompAccelW_ = sva::MotionVecd::Zero();
+
+  // Reset the target wrench to zero
+  targetWrench_ = sva::ForceVecd::Zero();
+  measuredWrench_ = sva::ForceVecd::Zero();
+  filteredMeasuredWrench_ = sva::ForceVecd::Zero();
+  lowPass_.reset(sva::ForceVecd::Zero());
+}
+
+void ImpedanceTask::load(mc_solver::QPSolver & solver, const mc_rtc::Configuration & config)
+{
+  if(config.has("impedanceM"))
+  {
+    impedanceM(config("impedanceM"));
+  }
+  if(config.has("impedanceD"))
+  {
+    impedanceD(config("impedanceD"));
+  }
+  if(config.has("impedanceK"))
+  {
+    impedanceK(config("impedanceK"));
+  }
+  if(config.has("wrenchGain"))
+  {
+    wrenchGain(config("wrenchGain"));
+  }
+
+  if(config.has("target"))
+  {
+    desiredPose(config("target"));
```

Comment:
Maybe we should remove this part to remove the conflict with 
https://github.com/jrl-umi3218/mc_rtc/pull/100/files#diff-1ea7af2780604d3059658bfd0c8f69e98c1a603be1810ec9c1828df358f272a1R205

Reply:
Nice catch, done in 9558f32